types/src/dataio.rs

```diff
@@ -19,5 +21,5 @@ pub trait DataProvider<Data>: Sync + Send + 'static {
 pub trait FinalizationHandler<Data>: Sync + Send + 'static {
     /// Data, provided by [DataProvider::get_data], has been finalized.
     /// The calls to this function follow the order of finalization.
-    fn data_finalized(&mut self, data: Data);
+    fn data_finalized(&mut self, data: Data, creator: NodeIndex);
```

Comment:
Since you are changing the API you got to bump the (minor because 0.x) version here, and in `consensus` at least.